CanvasPlusPlayground/Views/CalendarView.swift

```diff
@@ -0,0 +1,31 @@
+//
+//  CalendarView.swift
+//  CanvasPlusPlayground
+//
+//  Created by Jiyoon Lee on 9/17/24.
+//
+
+import SwiftUI
+
+struct CalendarView: View {
```

Comment:
This should just retrieve an `ics` for a single course, so there's no need to fetch all courses again. It would look something like this:

```swift
struct CalendarView: View {
    let course: Course

    init(course: Course) {
        self.course = course
    }

    var body: some View {
        Group {
            if let icsURL = course.calendar?.ics {
                Text(icsURL)
            } else {
                Text("No calendar available")
            }
        }
        .navigationTitle("Calendar")
    }
}
```

And then we need to somehow extract the events from the `.ics` file. I found [this Swift package](https://github.com/ramonvasc/MXLCalendarManagerSwift) that does this, if you can find something better, go for it.

Reply:
I agree here. Just use the `courseManager` injected as an @Environment. No need to recreate it. We can also just render the calendar information on a per-course basis.

Reply:
Fetching the ics file and parsing it might be overkill for now IMO. For now, it would suffice if you just redirect the user to the URL for that ics file, for them to download. Thoughts @rahulon12 ?

Reply:
Yeah that's fine for now.

> I agree here. Just use the `courseManager` injected as an @Environment. No need to recreate it. We can also just render the calendar information on a per-course basis.

You shouldn't need courseManager at all in `CalendarView.swift`, the ics link is part of the `Course` object

Reply:
Oh yeah, that's right